### PR TITLE
Interactivity API: Add support for underscores and leading dashes in the suffix part of the directive

### PR DESCRIPTION
**Pull request by cbravobernal**
## What?

Allow data-wp-class to contain classes with underscores, like BEM notation.

## Why?

Cause WordPress is using this notation for their classes, and, trying to migrate Search block to the Interactivity API, I found that the directive was not processed for the class that hides the search input.

## How?

Changing REGEX 😭 

## Testing Instructions

I added a couple of tests, but still, feel free to test it by using `data-wp-class--underscores__for_the_win` in an interactive block and check that everything is working as expected.

### CODE REVIEW

File: packages/interactivity/src/vdom.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ const directiveParser = new RegExp(
 		// (Optional) Match '--' followed by any alphanumeric charachters. It
 		// excludes underscore intentionally to prevent confusion, but it can
 		// contain multiple hyphens. E.g., "--custom-prefix--with-more-info".
-		'(?:--([a-z0-9][a-z0-9-]+))?$',
+		'(?:--([a-z0-9_-]+))?',
```
Comment (luisherranz) [inline packages/interactivity/src/vdom.js]: Why did you remove the final `$`?

**Review by luisherranz — APPROVED** [review]
Codewise looks good to me. 

Can you please add an entry to the changelog?

Also, is the removal of the final `$` intended?

File: packages/interactivity/src/vdom.js (hunk 1 above)
Comment (cbravobernal) [reply packages/interactivity/src/vdom.js]: Copy-pasted error. Is not affecting tests, so better leave it as it was.

Comment (luisherranz) [reply packages/interactivity/src/vdom.js]: Great, thank you!
